huxley/www/js/components/ChairSummaryView.js

```diff
@@ -0,0 +1,185 @@
+  mixins: [
+    ReactRouter.History,
+  ],
+
+  getInitialState() {
+    var user = CurrentUserStore.getCurrentUser();
+    return {
+      assignments: AssignmentStore.getCommitteeAssignments(user.committee),
+      countries: CountryStore.getCountries(),
+      country_assignments: {},
+      delegates: DelegateStore.getCommitteeDelegates(user.committee),
+    };
+  },
+
+  componentWillMount() {
+    var user = CurrentUserStore.getCurrentUser();
+    if (!User.isChair(user)) {
+      this.history.pushState(null, '/');
+    }
+  },
+
+  componentDidMount() {
+    var user = CurrentUserStore.getCurrentUser();
+    this._handleMapAssignments();
+    this._delegatesToken = DelegateStore.addListener(() => {
+      this.setState({delegates: DelegateStore.getCommitteeDelegates(user.committee)});
+      this._handleMapAssignments();
+    });
+
+    this._assignmentsToken = AssignmentStore.addListener(() => {
+      this.setState({assignments: AssignmentStore.getCommitteeAssignments(user.committee)});
+      this._handleMapAssignments();
+    });
+
+    this._countriesToken = CountryStore.addListener(() => {
+      this.setState({countries: CountryStore.getCountries()});
+    });
+  },
+
+  componentWillUnmount() {
+    this._countriesToken && this._countriesToken.remove();
+    this._delegatesToken && this._delegatesToken.remove();
+    this._assignmentsToken && this._assignmentsToken.remove();
+  },
+
+  render() {
+    return (
+      <InnerView>
+        <h2>Summaries</h2>
+        <p>
+          Here you can provide feedback for the delegates. Note that publishing
+          the summaries will allow advisors to view them.
+        </p>
+        <form>
+          <div className="table-container">
+            <table className="table highlight-cells">
+              <thead>
+                <tr>
+                  <th>Assignment</th>
+                  <th>Summary</th>
+                </tr>
+              </thead>
+              <tbody>
+                {this.renderSummaryRows()}
+              </tbody>
+            </table>
+          </div>
+          <Button
+            color="green"
+            onClick={this._handleSaveSummaries}>
+            Save
+          </Button>
+          <Button
+            color="blue"
+            onClick={this._handlePublishSummaries}>
+            Publish
+          </Button>
+        </form>
+      </InnerView>
+    );
+  },
+
+  renderSummaryRows() {
+    var committeeCountries = Object.keys(this.state.country_assignments);
+    var countries = this.state.countries;
+    return committeeCountries.map(country => {
+      var country_delegates = this.state.country_assignments[country];
+      return (
+        <tr key={country}>
+          <td>
+            {Object.keys(countries).length ? countries[country].name : country}
+          </td>
+          <td>
+            <textarea
+              className="text-input"
+              style={{"width": "95%"}}
+              rows="3"
```

Comment:
Depending on feedback we get, this may or may not need to be one or two rows larger